code/modules/emotes/emote_define.dm

```diff
@@ -78,18 +80,29 @@
 		use_3p = replacetext(use_3p, "USER", "<b>\the [user]</b>")
 		use_3p = capitalize(use_3p)
 
-	if(ismob(user))
-		var/mob/M = user
-		if(message_type == AUDIBLE_MESSAGE)
-			M.audible_message(message = use_3p, self_message = use_1p, deaf_message = emote_message_impaired, checkghosts = /datum/client_preference/ghost_sight)
-		else
-			M.visible_message(message = use_3p, self_message = use_1p, blind_message = emote_message_impaired, checkghosts = /datum/client_preference/ghost_sight)
+	if(message_type == AUDIBLE_MESSAGE)
+		if(isliving(user))
+			var/mob/living/L = user
+			if(!L.silent)
+				user.audible_message(message = use_3p, self_message = use_1p, deaf_message = emote_message_impaired, checkghosts = /datum/client_preference/ghost_sight)
+			else
+				user.visible_message(message = "[user] noiselessly opens mouth!", self_message = "You cannot say anything!", blind_message = emote_message_impaired, checkghosts = /datum/client_preference/ghost_sight)
+	else
+		user.visible_message(message = use_3p, self_message = use_1p, blind_message = emote_message_impaired, checkghosts = /datum/client_preference/ghost_sight)
 
 	do_extra(user, target)
+	do_sound(user, target)
 
 /decl/emote/proc/do_extra(var/atom/user, var/atom/target)
 	return
 
+/decl/emote/proc/do_sound(var/atom/user)
+	if(emote_sound)
+		if(islist(emote_sound))
+			if(emote_sound[user.gender])
+				return playsound(user.loc, pick(emote_sound[user.gender]), 50, 0)
+		return playsound(user.loc, pick(emote_sound), 50, 0)
```

Comment:
It looks like this should be tabbed in one step.
You can also remove `if(emote_sound)` because `islist()` will fail if it is null/false.